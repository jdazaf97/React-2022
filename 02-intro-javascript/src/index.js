
//OBJETOS LITERALES

    //debemos tener en cuenta inicialmente que cuando vemos los corcheres o llaves '{}' en consola o en cualquier lugar se entiende que eso se refiere a un objeto literal (o diccionarios, depende del leguaje de programacion estos cambian el nombre), por lo que para definir un objeto se hace de la siguiente forma:

const persona1 = {
    nombre: 'Juan'
}

//NOTA: Los objetos trabajan en general con pares de valores, en donde tendremos en cuenta una llave y un valor: [llave]: '[valor]', como se ve en el ejemplo anterior de la constante persona.

//Llamamos a consola
console.log ( persona1 );

//Si unicamente quiero recuperar el valor especifico (en consola) del objeto se se hace el llamado con un punto seguido del nombre de la llave:
console.log ( persona1.nombre );




console.log('Hola Mundo!');
