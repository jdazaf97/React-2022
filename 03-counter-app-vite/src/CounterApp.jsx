
//importamos el Hook useState
import {useState} from 'react';

//importanto prop types
import PropTypes from 'prop-types';

//exportacion y contenido del counterApp
export const CounterApp = ( {value} ) => {

    const [ counter, setCounter ] = useState( value );
    
    //Aumenta tamano en 1
    const handleAdd = (event) => {
        // console.log(event);
        setCounter( counter + 1 );
    }

    //Disminuye handleRes
    const handleSbt = (event) => {
        setCounter( counter - 1 );
    }

    //Resetea el contador
    const handleRes = (event) =>{
        setCounter( value );
    }
    
    
    return (        
        <>
            <h1>CounterApp</h1>
            <h2>{ counter }</h2>
            <button onClick={ handleAdd }> +1 </button>
            <button onClick={ handleSbt }> -1 </button>
            <button aria-label='btn-reset' onClick={ handleRes }> Reset </button>
        </>
    )
}

//definiendo prop Types
CounterApp.protoTypes = {
    //condicion
    value: PropTypes.number.isRequired,
}