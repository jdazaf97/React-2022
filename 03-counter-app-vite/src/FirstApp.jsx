

//importar proptypes
import PropTypes from 'prop-types';

//esportacion y contenido de FirstApp
export const FirstApp = ( {title, subTitle, name} ) => {

  // console.log(props);
  
  return (

    <>
        <h1 data-testid='test-title'> { title } </h1>
        <p>{ subTitle }</p>
        <p>{ subTitle }</p>
        <p>{ name }</p>
        {/* <code>{ JSON.stringify(newMessage) }</code> */}
    </>
  )
}


//definiendo prop types
FirstApp.propTypes = {
  //primera condicion:
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string
}

//definiendo prop default
FirstApp.defaultProps = {
  name: 'Juan felipe Daza',
  // title: 'No hay titulo',
  subTitle: 'No hay subtitulo'
}