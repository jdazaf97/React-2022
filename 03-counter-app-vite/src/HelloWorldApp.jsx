
export const HelloWorldApp = () => {
  return (
    <h1>HelloWorldApp</h1>//esto es lo que veremos en pantalla
  )
}

