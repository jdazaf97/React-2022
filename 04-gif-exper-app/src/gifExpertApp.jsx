
import { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = () => {

    const [categories, setCategories] = useState( ['Hollow Knigth', 'Stardew Valley'] );
    
    const onAddCategory = (newCategory) => {

        if( categories.includes(newCategory) ) return;

        // categories.push(newCategory)

        //Usando el operador spread podemos agregar una categoria teniendo en cuenta las anteriores puestas, sin emnargo esto viola el hecho que no hay una key unica:
        setCategories([ ...categories, newCategory ]);
        // setCategories([  newCategory, ...categories ]);
        
    };

    return (

        <>
            <h1>GifExpertApp</h1>
        

            <AddCategory 
                onNewCategory = { event => onAddCategory(event) }
            />

            {    
            categories.map( (category) => (
                    <GifGrid key={category} category={ category } /> 
                )) 
            }   
            
        </>

    )

}