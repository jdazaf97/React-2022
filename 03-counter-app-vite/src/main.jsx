
//instalamos la importacion de react
import React from 'react';

//tambien la importacion del client
import ReactDOM from 'react-dom/client';

//Importacion del FirstApp:
import { FirstApp } from './FirstApp';

//Importacion del counterApp:
import { CounterApp } from './CounterApp';

//aqui vamos a importar una hoja de estilos (CSS) o estilos globales:
import './style.css';

//importacion del HelloWorldApp
// import { HelloWorldApp } from './HelloWorldApp';
   

//Ahora procedemos a renderizar:

//en este caso dentro del HTML queremos ingresar esto dentro del cuerpo de este y en un div con id 'root'
ReactDOM.createRoot( document.getElementById('root') ).render(
    <React.StrictMode>
        <CounterApp value={100} />
        {/* <FirstApp title='Hola Soy Felipe' /> */}
    </React.StrictMode>
)
