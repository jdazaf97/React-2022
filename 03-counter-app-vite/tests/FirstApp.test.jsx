import { render } from '@testing-library/react';
import { FirstApp } from '../src/FirstApp';



describe('Pruebas en <FirstApp />', () => { 

    test('Debe hacer match con el SnapShot', () => { 

        const tittle = 'Hola soy Juan';
        const { container } = render( <FirstApp title={ tittle }/> );
        
        expect( container ).toMatchSnapshot();
        
    })
    
    test('Debe de mostrar el titulo en un h1', () => { 

        const title = 'Hola soy Juan';
        const { container, getByText, getByTestId } = render( <FirstApp title={ title }/> );

        expect( getByText(title) ).toBeTruthy();

        // const h1 = container.querySelector('h1')
        expect( getByTestId('test-title').innerHTML ).toContain(title);

     })

     test('Debe mostrar el subtitulo enviado por props', () => { 

        const title = 'Hola soy Juan';
        const subTitle = 'Soy el subtitulo';
        const { getAllByText } = render( 
            <FirstApp 
                title={ title }
                subTitle={ subTitle }
            /> 
        );

        expect( getAllByText(subTitle).length ).toBe(2);
        
      })
    
        


})
