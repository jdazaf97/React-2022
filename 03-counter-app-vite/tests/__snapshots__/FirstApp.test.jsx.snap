// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> Debe hacer match con el SnapShot 1`] = `
<div>
  <div
    data-testid="test-title"
  >
     
    Hola soy Juan
     
  </div>
  <p>
    No hay subtitulo
  </p>
  <p>
    No hay subtitulo
  </p>
  <p>
    Juan felipe Daza
  </p>
</div>
`;
